htlcswitch/link.go

```diff
@@ -364,6 +365,10 @@ type channelLink struct {
 	// resolving those htlcs when we receive a message on hodlQueue.
 	hodlMap map[models.CircuitKey]hodlHtlc
 
+	// flushCont is a function that is called when the channel finishes
+	// flushing.
+	flushCont fn.MVar[func()]
```

Comment:
Still getting through the diff, but: if this is always managed in the main event loop of the channel, then do we need this new concurrent-safe swappable condition variable? Otherwise I'm imagining something a bit simper (and a pattern we've used elsewhere), where we store the set of subscribers to notify after some condition/state has been met. 

Still trying to piece together the final control flow here....

Reply:
Because Flush is on the link interface we can't guarantee that we won't have callsites in different threads.